### PR TITLE
[13.0][MIG] utm

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Migration scripts of `utm`.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kos94ok-3D** [comment]
I think need to merge mail.mass_mailing.campaign and utm.campaign models
https://github.com/odoo/odoo/commit/a661b00015557281faca7c88eae4aa5321a2b843#

**Comment by MiquelRForgeFlow** [comment]
Thanks @kos94ok-3D , it slipped my mind.

File: addons/utm/migrations/13.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from openupgradelib import openupgrade
+
+_model_renames = [
+    ('mail.mass_mailing.campaign', 'utm.campaign'),
+    ('mail.mass_mailing.stage', 'utm.stage'),
+    ('mail.mass_mailing.tag', 'utm.tag'),
+]
+
+_table_renames = [
+    ('mail_mass_mailing_campaign', 'utm_campaign'),
```
Comment (kos94ok-3D) [inline addons/utm/migrations/13.0.1.0/pre-migration.py]: I don`t sure that it correctly because utm_campaign exists in v12

**Comment by MiquelRForgeFlow** [comment]
Yes, in v12, `mail.mass_mailing.campaign` has `_inherits = {'utm.campaign': 'campaign_id'}`

**Comment by pedrobaeza** [comment]
Then the code should be more sophisticated (I will check any way this later).

**Comment by MiquelRForgeFlow** [comment]
yeah, I am working on it

**Comment by MiquelRForgeFlow** [comment]
It's ready. @kos94ok-3D could you review again? :)

**Comment by kos94ok-3D** [comment]
Looks good. Just code review.

**Review by pedrobaeza — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
utm does not depend on mass_mailing so scripts fail when mass_mailing is not installed previously :scream:

**Comment by pedrobaeza** [comment]
It shouldn't. Please put the stack trace that supports your affirmation.

**Comment by AaronHForgeFlow** [comment]
This is the stack trace:

```
2020-08-03 13:47:19,864 7982 INFO database odoo.modules.migration: module utm: Running migration [>13.0.1.0] pre-migration 
2020-08-03 13:47:20,138 7982 INFO database OpenUpgrade: utm: pre-migration script called with version 12.0.1.0 
2020-08-03 13:47:20,459 7982 INFO database odoo.modules.registry: module utm: creating or updating database tables 
2020-08-03 13:47:20,476 7982 ERROR database odoo.sql_db: bad query: SELECT "utm_stage".id FROM "utm_stage" ORDER BY "utm_stage"."sequence"   limit 1
ERROR: relation "utm_stage" does not exist
LINE 1: SELECT "utm_stage".id FROM "utm_stage" ORDER BY "utm_stage"....
```

The renames for model, table and xmlid are not executed (that is correct), but then it fails when upgrading the module. It is weird because it tries to read a table before creating it. I will update this comment with my findings.

**Comment by pedrobaeza** [comment]
It shouldn't fail, as Odoo will create the table for you in most cases.

**Comment by AaronHForgeFlow** [comment]
I think the problem is Odoo defines a relation to utm_stage  https://github.com/odoo/odoo/blob/13.0/addons/utm/models/utm.py#L27 before defining the model https://github.com/odoo/odoo/blob/13.0/addons/utm/models/utm.py#L51 so no issue with Openupgrade

Edited: Confirmed. It does not fail if I move that code upwards. Thank you @pedrobaeza

**Pull request merged**